### PR TITLE
Scherer

### PR DESCRIPTION
**Pull request by schererc**
- changed spline.h and cubicspline.h routines to deal with periodic boundary conditions and additional constraints necessary for tabulated dihedral potentials

- changed qrsolve.cc to deal with arbitrary number of constraints

### CODE REVIEW

File: netbeans/libtools/Makefile_nb [hunk 1]
```diff
@@ -1,92 +0,0 @@
-#
```
Comment (junghans) [inline netbeans/libtools/Makefile_nb]: Why?

**Comment by junghans** [comment]
Two comments:
- Copyright headers need update
- Why did you remove the netbeans Makefiles?

**Comment by junghans** [comment]
Build failed due to https://travis-ci.org/votca/csg/jobs/156518043#L4324
```
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Failed
```

**Comment by junghans** [comment]
Build failed due to https://travis-ci.org/votca/csg/jobs/156518043#L4324
```
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Failed
```

File: netbeans/libtools/Makefile_nb (hunk 1 above)
Comment (schererc) [inline netbeans/libtools/Makefile_nb]: I removed the whole netbeans folder. Probably not the best idea.

Comment (junghans) [inline netbeans/libtools/Makefile_nb]: I reverted that in schererc/tools#1, so merge that first!

**Pull request merged**